.github/workflows/ci.yml

```
@@ -1,168 +1,218 @@
+  NODE_VERSION: '18'
+  MADSPARK_MODE: mock
 
 jobs:
+  # Quick validation phase - fail fast on basic issues
+  quick-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      
+      - name: Set up Python
+        uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)